Xamarin.PropertyEditing.Mac/PropertyTableDelegate.cs

```diff
@@ -109,23 +113,30 @@ PropertyEditorControl GetEditor (EditorViewModel vm, NSOutlineView outlineView)
 			Type propertyType = vm.GetType ();
 			if (!ViewModelTypes.TryGetValue (propertyType, out controlType)) {
 				if (propertyType.IsConstructedGenericType) {
-					genericArgs = propertyType.GetGenericArguments ();
+					genericArgs = GetCorrectGenericArguments (propertyType.GetGenericArguments ());
 					propertyType = propertyType.GetGenericTypeDefinition ();
 					ViewModelTypes.TryGetValue (propertyType, out controlType);
 				}
 			}
 
-			if (controlType == null)
-				return null;
+			if (controlType != null) {
+				if (controlType.IsGenericTypeDefinition) {
+					controlType = controlType.MakeGenericType (genericArgs);
+				}
 
-			if (controlType.IsGenericTypeDefinition) {
-				if (genericArgs == null)
-					genericArgs = propertyType.GetGenericArguments ();
+				return SetUpEditor (controlType, vm, outlineView);
+			} else
+				return null;
+		}
 
-				controlType = controlType.MakeGenericType (genericArgs);
+		private Type[] GetCorrectGenericArguments (Type[] types)
```

Comment:
This shouldn't be needed, as it stands the properties VM will treat all numerics as nullables, so the controls should just deal with them as such. That's why `Numeric` can handle `T` as a `Nullable<>`

Reply:
@ermau This is needed at the point where we are dynamically creating the control, before we insert it into the TableView. 
`controlType.MakeGenericType (genericArgs);` will fail/crash if we try to create the type with a Nullable<>. If you know of a way around that, then I'm all ears.

Reply:
Because you're constraining it to a struct, `Nullable` isn't a struct.

Reply:
@ermau I think I misunderstand what you mean, because according to the docs `Nullable<T>` is a struct - https://msdn.microsoft.com/en-us/library/b3h38hb0(v=vs.110).aspx

Reply:
No, I'm just misremembering an error I ran into related to this before. I need the crash info if I'm to help instead of just guessing.

Reply:
I think this is all the info you might need - https://gist.github.com/CartBlanche/34a918b06d77fabdfec87112088ad22b
Let me know if I'm missing any info you need.